src/magic_folder/testing/web.py

```diff
@@ -266,16 +286,23 @@ def _add_entry_to_dir(self, request, dircap, segments):
 
         kind = "dirnode" if IDirnodeURI.providedBy(content) else "filenode"
 
-        # so it looks like Tahoe returns "mutable=true" even if you
-        # only have a read-only capability ..
+        # Objects can be mutable or immutable.  Capabilities can be read-only
+        # or write-only.  Don't mix up mutability with writeability.  You
+        # might have a read-only cap for a mutable object.  In this case,
+        # someone *else* might be able to change the object even though you
+        # can't.
         metadata = {
             "mutable": content.is_mutable(),
             "ro_uri": content_cap,
-            "verify_cap": content.get_verify_cap().to_string(),
+            "verify_uri": content.get_verify_cap().to_string(),
+            "format": _get_node_format(content),
         }
         if content_cap != content.get_readonly().to_string():
             metadata["rw_uri"] = content_cap
 
+        if kind == "filenode" and content.get_size() is not None:
+            metadata["size"] = content.get_size()
```

Comment:
Not sure if it helps or not, but the size of immutable things will be encoded in the capability itself.

Reply:
Yea, I noticed that too.  This change is only meant to make the response look more like the Tahoe-LAFS response.  It looks like Tahoe-LAFS doesn't include the size for immutable directories, only immutable files.